### PR TITLE
Attach Auth token to header on every HTTP request

### PR DESCRIPTION
**Pull request by nakhbari**
Issue: #752 

Depends on #781 

- [x] Attach JWT to all outgoing HTTP requests

### CODE REVIEW

File: web/app/services/auth.interceptor.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  intercept(req: HttpRequest<any>, next: HttpHandler):
+      Observable<HttpEvent<any>> {
+    const token = localStorage.getItem(LocalStorageKeys.AUTH_TOKEN);
+
+    // Attach auth token to request if it exists. Otherwise, handle it normally.
+    if (token) {
+      const authorizedRequest = req.clone(
+          {headers: req.headers.set('Authorization', `Bearer ${token}`)});
```
Comment (minuscorp) [inline web/app/services/auth.interceptor.ts]: might receive always a `403` when the token has expired, maybe this `Interceptor` should try to fetch a new token?

Comment (nakhbari) [reply web/app/services/auth.interceptor.ts]: eventually i'd like to store the expiry of the token to determine if the user is still logged in. If the token is expired we should block the requests, until they log in.

Comment (taquitos) [reply web/app/services/auth.interceptor.ts]: @nakhbari let's create an issue to track this idea 👍

Comment (nakhbari) [reply web/app/services/auth.interceptor.ts]: it's one of the subtasks in #752

**Pull request merged**